cosmos/dbt/selector.py

```diff
@@ -148,22 +150,38 @@ def filter_nodes(self, nodes: dict[str, DbtNode]) -> set[str]:
         :return: set of node ids that matches current graph selector
         """
         selected_nodes: set[str] = set()
+        root_nodes: set[str] = set()
 
         # Index nodes by name, we can improve performance by doing this once
         # for multiple GraphSelectors
-        node_by_name = {}
-        for node_id, node in nodes.items():
-            node_by_name[node.name] = node_id
+        if PATH_SELECTOR in self.node_name:
+            path_selection = self.node_name[len(PATH_SELECTOR) :]
+            root_nodes.update({node_id for node_id, node in nodes.items() if path_selection in str(node.file_path)})
+
+        elif TAG_SELECTOR in self.node_name:
+            tag_selection = self.node_name[len(TAG_SELECTOR) :]
+            root_nodes.update({node_id for node_id, node in nodes.items() if tag_selection in node.tags})
+
+        elif CONFIG_SELECTOR in self.node_name:
```

Comment:
Please, could we add a test covering this selector - so we make sure it continues working as expected after the refactor?

Reply:
added these test cases here: https://github.com/astronomer/astronomer-cosmos/blob/67cb56ff9c4d7ab20dcd3da926a83aa83bea80b6/tests/dbt/test_selector.py#L490-L505